src/components/web3-connect/index.tsx

```diff
@@ -156,6 +156,16 @@ export class Container extends React.Component<Properties, State> {
     if (web3.account !== previouslyAccount) {
       this.props.setAddress(web3.account);
     }
```

Comment:
we should really refactor this code, it's pretty hard to understand. Plus i've added one more condition 😂